### PR TITLE
CAP-0035: Changes from the Open Protocol Meeting held on 2021-01-28

### PR DESCRIPTION
**Pull request by sisuresh**
This PR has a few changes -

1. Change `SetTrustLineFlagsOp` `AssetCode` parameter to `Asset` so we can extend use of this operation in the future.
2. Add additional validation conditions for the `Asset` parameter specified above.
3. Expand existing explanations.

### CODE REVIEW

File: core/cap-0035.md [hunk 1]
```diff
@@ -268,7 +271,7 @@ index 7f08d757..ebbf45f5 100644
 +struct SetTrustLineFlagsOp
 +{
 +    AccountID trustor;
-+    AssetCode asset;
++    Asset asset;
```
Comment (leighmcculloch) [inline core/cap-0035.md:274]: ❗ The same change needs making to `ClawbackOp`. We agreed to also defer who can issue the `ClawbackOp` for the same reasons, that an issuer may be able to delegate that responsibility in the future at which point the operation must encode the issuer to reference the asset.

File: core/cap-0035.md [hunk 2]
```diff
@@ -627,6 +633,8 @@ A new operation, `SetTrustLineFlagsOp`, is proposed in this cap to make
 trustline flag modification simpler. Additionally the `AllowTrustOp` operation is
 semantically intended to change authorization and clawback is outside it's scope.
 
+The `AllowTrustOp` is being deprecated for `SetTrustLineFlagsOp`.
```
Comment (leighmcculloch) [inline core/cap-0035.md:636]: 💡 
```suggestion
The `AllowTrustOp` is deprecated for `SetTrustLineFlagsOp`.

```

File: core/cap-0035.md [hunk 3]
```diff
@@ -87,6 +87,9 @@ The `ClawbackOp` may result in revocation of some or all of the specified
 assets from the designated account based on the amount provided in the
 operation.
 
+The `SetTrustLineFlagsOp` will allow the user to set and clear specific trustline
```
Comment (leighmcculloch) [inline core/cap-0035.md:90]: 💡 
```suggestion
The `SetTrustLineFlagsOp` allows the user to set and clear specific trustline 
```

File: core/cap-0035.md [hunk 4]
```diff
@@ -644,6 +652,12 @@ allow an issuer to disable clawback on an asset entirely by removing `AUTH_CLAWB
 on its own account and clearing `TRUSTLINE_CLAWBACK_ENABLED_FLAG` from all existing trustlines to 
 that asset.
 
+`SetTrustLineFlagsOp` uses an `Asset` as a parameter instead of an `AssetCode`. This won't matter
+in protocol 16 because the operation is invalid if the issuer on `Asset` is not the source account,
+but it'll allow us to explore options for an account that's not the issuer to change trustline flags.
+The two scenarios we currently have in mind are the owner changing the trustline flag and the issuer
+delegating trustline flag modification to a different account.
```
Comment (leighmcculloch) [inline core/cap-0035.md:659]: 💡 Suggest rewording to focus on deferring the decision rather than catering to possible future decisions, but if you prefer the existing language I'm still 👍🏻 with it.
```suggestion
`SetTrustLineFlagsOp` uses an `Asset` as a parameter instead of an `AssetCode` to defer the decision of who all future authorizers of the operation can be. The `Asset` includes the issuer account, where-as use of `AssetCode` would assume the source account is always the issuer account. In this proposal, only the issuer may authorize the operation. This decision makes it possible in a future proposal to expand the accounts that may be allowed to issue the operation.
```

**Review by leighmcculloch — COMMENTED** [review]
See below at one change (❗) that is missing, otherwise looks great!

I also left some rewording suggestions (💡) to keep the tone of the document consistent and to focus on indicating the intention to defer decisions rather than make them for future use cases.

File: core/cap-0035.md (hunk 1 above)
Comment (sisuresh) [reply core/cap-0035.md:274]: Ah I missed this during the discussion but you're right. Fixed.

**Review by leighmcculloch — APPROVED** [review]
